### ISSUE TITLE
installed.list

### ISSUE THREAD

**Issue by illori** [SimpleMachines/SMF#775]
why do we need this file? can it be safely be removed? some people in the past have deleted it by accident and then their install tells them that all their packages are uninstalled where in the db they are still listed as installed.

**Comment by Arantor** [comment]
It's mostly legacy; back in 1.1 days it was the only method of knowing what was installed and what wasn't.

In 2.0+ though its sole content is a timestamp indicating the last change to packages - and if it's gone it wipes out the list of installed.

I have no idea why we still have it but there _must_ be a reason. I just wish I knew what it was.

**Comment by illori** [comment]
if no reason is found, can we get rid of it?

**Comment by Arantor** [comment]
FYI: I went back and looked through the entire old SVN history surrounding this change, it doesn't seem to be documented anywhere, but as far as I can tell it was implemented by Grudge like 8 years ago, around the time that support for 'install to other themes' was added, which was all the guts of log_packages being used. While it does provide a failsafe for flagging everything uninstalled, that's really not ideal as we've found.

There are several places that use it to 'flush' the package list, except at no point is the contents actually tested for a last-modified time; the only time it is actually checked for content is to determine if it is empty and if so, nuke the installed mods list.

The one side benefit of any use, not that it IS any use, is for an is_writable test in Packages/, but that's no guarantee that anything will be workable anyway.
